tests/prompts_test.py

```diff
@@ -0,0 +1,362 @@
+                                    f"Testcase-{index_of_testcase}: "
+                                    f"Expected: {expected_result} "
+                                    f"LLM: {sentence}\n\n"
+                                )
+                                print(
+                                    f"Circle-{index_of_circle + 1} "
+                                    f"Claude: Test-{index_of_testcase} FAILED: "
+                                    f"Result: {sentence}"
+                                )
+                            result_testing_tuples.append(
+                                (
+                                    index_of_circle + 1,
+                                    index_of_testcase,
+                                    expected_result,
+                                    sentence,
+                                    llm_response,
+                                    CLAUDE_PROMPT_TEMPLATE,
+                                    INSTRUCTION,
+                                    OPTIONS,
+                                    CLAUDE_MODEL,
+                                )
+                            )
+
+        self.save_response_to_csv(
+            result_testing_tuples, "claude"
+        )  # Save prompts and llm responses to csv file
+
+        accuracy = round(match_count / (len(self.testcases) * RETRY_COUNT), 2)
+        print(f"Accuracy is {accuracy}")
+
+        self.assertGreaterEqual(
+            accuracy,
+            CLAUDE_ACCURACY,
+            f"The match occurred only {match_count} "
+            f"times out of {len(self.testcases) * RETRY_COUNT} \n{failed_testcases}",
+        )
+
+    def test_qwen_prompt(self):
+        match_count = 0
+        failed_testcases = ""
+        result_testing_tuples = []
+        client = OpenAI(
+            base_url=OPENROUTER_BASE_URL,
+            api_key=OPENROUTER_API_KEY,
+        )
+        for index_of_circle in range(
+            RETRY_COUNT
+        ):  # Retry tests for correct evaluation precision
+            for index_of_testcase, expected_result, specification in self.testcases:
+                with self.subTest(i=index_of_testcase):  # subTest for every llm call
+                    completion = client.chat.completions.create(
+                        model=QWEN_MODEL,
+                        temperature=0,
+                        max_tokens=1000,
+                        messages=[
+                            {
+                                "role": "system",
+                                "content": QWEN_PROMPT_TEMPLATE.format(
+                                    OPTIONS=OPTIONS,
+                                    INSTRUCTION=INSTRUCTION,
+                                ),
+                            },
+                            {
+                                "role": "user",
+                                "content": specification,
+                            },
+                        ],
```

Comment:
this logic is very similar for each test, why not to extract it into one function.
I believe we also have similar code in openrouter.py, so I would reuse code from there instead of duplicating it.

Reply:
Made this in one function, added loom with explanation